src/main/java/seedu/address/storage/bidstorage/JsonAdaptedBid.java

```diff
@@ -50,12 +52,16 @@ public Bid toModelType() throws IllegalValueException {
         if (propertyId == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, "P12"));
         }
+        PropertyId modelPropertyId = new PropertyId(propertyId);
+
         if (bidderId == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, "B04"));
         }
+        BidderId modelBidderId = new BidderId(bidderId);
+
```

Comment:
whats the point of getting the .toString if you are just gonna put it pack into the modelPropertyId. Why not just check if the propertyId of type PropertyId instead of String contains null then it throw exception

Reply:
like still can but idk if considered redundant or not

Reply:
I am not sure exactly how these json creators work but I think they need to be stored as a String because of the json format. And it is the same format as the original AB3's implementation. I am not very sure though.